x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptLongMappedFieldType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.index.mapper.NumberFieldMapper.NumberType;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.xpack.runtimefields.LongScriptFieldScript;
+import org.elasticsearch.xpack.runtimefields.fielddata.ScriptLongFieldData;
+import org.elasticsearch.xpack.runtimefields.query.LongScriptFieldExistsQuery;
+import org.elasticsearch.xpack.runtimefields.query.LongScriptFieldTermQuery;
+
+import java.util.Map;
+
+public class ScriptLongMappedFieldType extends AbstractScriptMappedFieldType {
+    private final LongScriptFieldScript.Factory scriptFactory;
+
+    ScriptLongMappedFieldType(String name, Script script, LongScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, script, meta);
+        this.scriptFactory = scriptFactory;
+    }
+
+    @Override
+    protected String runtimeType() {
+        return "long";
+    }
+
+    @Override
+    public Object valueForDisplay(Object value) {
+        return value; // These should come back as a Long
+    }
+
+    @Override
+    public ScriptLongFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
+        // TODO once we get SearchLookup as an argument, we can already call scriptFactory.newFactory here and pass through the result
+        return new ScriptLongFieldData.Builder(scriptFactory);
+    }
+
+    private LongScriptFieldScript.LeafFactory leafFactory(QueryShardContext context) {
+        return scriptFactory.newFactory(script.getParams(), context.lookup());
+    }
+
+    @Override
+    public Query existsQuery(QueryShardContext context) {
+        checkAllowExpensiveQueries(context);
```

Comment:
should we introduce some way to not forget the check? like having this done in the base class, and define queries through new abstract methods?

Reply:
That is tricky because not all subclasses will override all queries......

I wonder if we can wait on this one until we have more subclasses.

Reply:
I see, it is fine to wait. Though this will require us to extend every single query at least for this check and then call super? Maybe it is actually a reason to force overriding every query? Not sure though

Reply:
know what? I am having second thoughts about this check and the way it is performed. We are going to fail on all shards which is not desirable. We should rather have a way to fail on the coordinating node, but I know that is hard. Maybe we should reconsider and discuss this?

Reply:
let's leave this for now, we will revisit this later